### PR TITLE
[Serve] Mutliplexed information report impl

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

- Multiplexed metrics.
- Pass multiplexed information into controller and make it available at `RunningReplicaInfo`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_multiplex.py [hunk 1]
```diff
@@ -21,22 +44,26 @@ async def model_load_func(model_id: str):
         assert multiplexer.models == {"1": "1"}
         # Load model2
         await multiplexer.load_model("2")
+        assert multiplexer._push_multiplexed_replica_info is True
```
Comment (edoakes) [inline python/ray/serve/tests/test_multiplex.py:47]: nit
```suggestion
        assert multiplexer._push_multiplexed_replica_info
```

Comment (edoakes) [reply python/ray/serve/tests/test_multiplex.py:47]: we should also have tests that this is not changed when models are not loaded/unloaded

File: python/ray/serve/multiplex.py [hunk 2]
```diff
@@ -94,3 +147,22 @@ async def unload_model(self) -> None:
             else:
                 await sync_to_async(model.__del__)()
             setattr(model, "__del__", lambda _: None)
+
+    async def _push_model_ids(self):
```
Comment (edoakes) [inline python/ray/serve/multiplex.py:151]: Is it possible to piggy-back this onto the existing messages that record the autoscaling stats?

File: python/ray/serve/tests/test_multiplex.py (hunk 1 above)
Comment (sihanwang41) [reply python/ray/serve/tests/test_multiplex.py]: done!

File: python/ray/serve/multiplex.py (hunk 2 above)
Comment (sihanwang41) [reply python/ray/serve/multiplex.py:151]: synced offline, keep as it is right now.

File: python/ray/serve/multiplex.py [hunk 3]
```diff
@@ -72,13 +119,19 @@ async def load_model(self, model_id: str) -> Any:
...
                 await self.unload_model()
+                self.model_unload_latency_s.set(time.time() - unload_start_time)
             # Load the model.
             logger.info(f"Loading model '{model_id}'.")
+            load_start_time = time.time()
             if self.self_arg is None:
                 self.models[model_id] = await self._func(model_id)
             else:
                 self.models[model_id] = await self._func(self.self_arg, model_id)
+            self._push_multiplexed_replica_info = True
```
Comment (zcin) [inline python/ray/serve/multiplex.py:138]: Do you have to set `_push_multiplexed_replica_info` to True in `unload_models`?

Comment (sihanwang41) [reply python/ray/serve/multiplex.py:138]: When there is cache miss, we will load new models and set the _push_multiplexed_replica_info to True.
Unload model is the intermediate step for loading new models.

**Comment by sihanwang41** [comment]
^^ @edoakes feel free to take another look, comments all resolved :)

**Review by edoakes — APPROVED** [review]

**Pull request merged**